src/poller.rs

```diff
@@ -1,38 +1,91 @@
-use std::path::Path;
+use std::fs;
+use std::path::{Path, PathBuf};
 use std::process;
 use std::time::Instant;
 
 use tokio::time;
 use tracing::{error, info};
 
-use crate::config::Config;
+use crate::config::{Config, WatchConfig};
 use crate::log::Operation;
 use crate::snapshots;
 
+/// Checks the provided `child_path` is a directory.
+/// If either `includes` or `excludes` are set,
+/// checks whether the path is included/excluded respectively.
+fn is_valid_directory(base_path: &Path, child_path: &PathBuf, value: &WatchConfig) -> bool {
```

Comment:
nit: I'm not an experienced Rustacean, but I think `child_path` should be a `&Path`, esp since you didn't need `&mut PathBuf`.